### PR TITLE
add AMDGPU support

### PR DESCRIPTION
**Pull request by radudiaconu0**
add AMDGPU extension support

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "julia.environmentPath": "/home/radu/NeuralAttentionlib.jl"
+}
```
Comment (chengchingwen) [inline .vscode/settings.json:3]: This file should be removed

File: Project.toml [hunk 2]
```diff
@@ -12,17 +12,20 @@ Static = "aedffcd0-7271-4cad-89d0-dc628f76c6d3"
...
-FiniteDifferences="26cc04aa-876d-5657-8c51-4c34ba976000"
+FiniteDifferences = "26cc04aa-876d-5657-8c51-4c34ba976000"
 Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f"
+AMDGPU = "21141c5a-9bdb-4563-92ae-f87d6854732e"
 
 [extensions]
 NeuralAttentionlibCUDAExt = "CUDA"
 NeuralAttentionlibFiniteDifferences = "FiniteDifferences"
 NeuralAttentionlibZygoteExt = "Zygote"
+NeuralAttentionlibCAMDGPUExt = "AMDGPU"
```
Comment (chengchingwen) [inline Project.toml:23]:
```suggestion
NeuralAttentionlibAMDGPUExt = "AMDGPU"
```

**Review by chengchingwen — COMMENTED** [review]
There are too many noises (e.g. small style changes) in this PR. I'll make a few modifications to this PR

File: .vscode/settings.json (hunk 1 above)
Comment (radudiaconu0) [reply .vscode/settings.json:3]: Yes. sorry my bad can you test it before merging? I didn't test it that much.

Comment (chengchingwen) [reply .vscode/settings.json:3]: I don't have any device with amdgpu, so I can only test if the code compiled without issues.

**Pull request closed without merging**